sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatDataWriter.scala

```diff
@@ -243,6 +273,52 @@ class DynamicPartitionDataWriter(
     statsTrackers.foreach(_.newFile(currentPath))
   }
 
+  /**
+   * Increase the file counter and open a new OutputWriter.
+   * This is used when number of records records exceeding limit.
+   *
+   * @param partitionValues the partition which all tuples being written by this `OutputWriter`
+   *                        belong to
+   * @param bucketId the bucket which all tuples being written by this `OutputWriter` belong to
+   */
+  protected def increaseFileCounter(
```

Comment:
shall we put the `if (... recordsInFile >= description.maxRecordsPerFile)` in this method as well? Then the method can be named as `renewCurrentWriterIfTooManyRecords`

Reply:
In `DynamicPartitionDataConcurrentWriter.write`, I need this condition check to update writer in map as well (as the current writer is renewed). If we put the condition inside here, we need double check in `DynamicPartitionDataConcurrentWriter.write`. Shall we keep it as it is?

Reply:
ah, let's keep it as it is

Reply:
let's still rename it: `renewCurrentWriterForTooManyRecords`